### PR TITLE
Changes for AIX compilation to get CPU of running thread. hz is inter…

### PR DESCRIPTION
**Pull request by madurais**
…nal variable in AIX, hence changing to hz1 in window_functions.cc so that all OS shall work

**Description**: Describe your changes.
6 files are changed. 
1. onnxruntime/core/common/logging/logging.cc 
In this file we have added code to include pthread.h for AIX with defined(_AIX)
2. onnxruntime/core/common/threadpool.cc
In this file changes to get the current CPU using mycpu() in AIX. 
3. onnxruntime/core/providers/cpu/signal/window_functions.cc
In AIX hz is an inbuilt variable. Hence changing the variable hz in the code to hz1. I have not checked the OS using defined(_AIX), as hz1 works for all OSes. 
4. onnxruntime/core/platform/posix/stacktrace.cc
In this file we added !defined(_AIX) in the below code:
#if !defined(__ANDROID__) && !defined(__wasm__) && !defined(_OPSCHEMA_LIB_) && !defined(_AIX)
#include <execinfo.h>
#endif
#include <vector>
5. onnxruntime/core/platform/path_lib.h
This file changed to enable to find file type in AIX with AIX system calls. 
6. onnxruntime/core/platform/posix/env.cc
Enable AIX similar to other platforms like _ANDROID_, __APPLE__, __wasm__

Please review and let me know if I miss something. Thanks. 

**Motivation and Context**
- Why is this change required? What problem does it solve?
      The current changes are for enabling compilation of the code in AIX Operating system. 
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by tvkai** [comment]
@snnn @tianleiwu @skottmckay @edgchen1 can you help us to get a review on this ??.

File: onnxruntime/core/platform/path_lib.h [hunk 1]
```diff
@@ -220,6 +224,66 @@
...
+template <typename T>
+void LoopDir(const std::string& dir_name, T func) {
+  DIR* dir = opendir(dir_name.c_str());
+  struct stat stats; 
+  if (dir == nullptr) {
+    auto e = errno;
+    char buf[1024];
+    char* msg;
+#if defined(__GLIBC__) && defined(_GNU_SOURCE) && !defined(__ANDROID__)
+    msg = strerror_r(e, buf, sizeof(buf));
```
Comment (skottmckay) [inline onnxruntime/core/platform/path_lib.h:257]: would __ANDROID__ ever be defined on AIX?

File: onnxruntime/core/providers/cpu/signal/window_functions.cc [hunk 2]
```diff
@@ -110,7 +110,7 @@ Status BlackmanWindow::Compute(OpKernelContext* ctx) const {
   return create_cosine_sum_window(ctx, data_type_, a0, a1, a2, is_periodic_);
 }
 
-static inline double hz_to_mel_scale(double hz) { return 2595 * std::log10(1 + hz / 700); }
+static inline double hz_to_mel_scale(double hz1) { return 2595 * std::log10(1 + hz1 / 700); }
```
Comment (skottmckay) [inline onnxruntime/core/providers/cpu/signal/window_functions.cc:114]: Can you explain how `hz` clashes with an AIX variable? Not clear to me how an argument name could do that.

Comment (madurais) [reply onnxruntime/core/providers/cpu/signal/window_functions.cc:114]: Hi @skottmckay 
In AIX /usr/include/sys/m_param.h, has 
#define __hz    HZ      /* Berkeley uses lower case hz      */
#define hz      __hz

This file is included in window_functions.cc through below include path:
/usr/include/sys/m_param.h <- /usr/include/sys/context.h <- /opt/freeware/lib/gcc/powerpc-ibm-aix7.3.0.0/10/include-fixed/sys/signal.h <- /opt/freeware/lib/gcc/powerpc-ibm-aix7.3.0.0/10/include-fixed/sys/wait.h <- /opt/freeware/lib/gcc/powerpc-ibm-aix7.3.0.0/10/include-fixed/stdlib.h <-  /opt/freeware/lib/gcc/powerpc-ibm-aix7.3.0.0/10/include/c++/cstdlib <- /opt/freeware/lib/gcc/powerpc-ibm-aix7.3.0.0/10/include/c++/bits/stl_algo.h <- /opt/freeware/lib/gcc/powerpc-ibm-aix7.3.0.0/10/include/c++/algorithm <- include/onnxruntime/core/common/common.h <- core/providers/cpu/signal/window_functions.h

File: onnxruntime/core/platform/path_lib.h (hunk 1 above)
Comment (madurais) [reply onnxruntime/core/platform/path_lib.h:257]: Hi @skottmckay, 
ANDROID not defined for AIX. I have removed that and pushed the changes to same branch. Thank you for catching it.

File: onnxruntime/core/providers/cpu/signal/window_functions.cc (hunk 2 above)
Comment (skottmckay) [reply onnxruntime/core/providers/cpu/signal/window_functions.cc:114]: Ah ok. Not the greatest choice of a #defne name.

Would be great to capture this reason for using 'hz1' in a comment so it's not accidentally changed in the future. As we don't have an AIX machine to build on our CIs would not fail if that happened.

Comment (madurais) [reply onnxruntime/core/providers/cpu/signal/window_functions.cc:114]: Yes - Will add that in comment. 
// 'hz' is a #define in AIX, hence using hz1. 
static inline double hz_to_mel_scale(double hz1) { return 2595 * std::log10(1 + hz1 / 700); }

One comment in first usage is sufficient right.

Comment (skottmckay) [reply onnxruntime/core/providers/cpu/signal/window_functions.cc:114]: I'd do it in both places. I wouldn't assume someone making a change in CreateMelWeightMatrix to have seen the comment in a completely separate function.

Comment (madurais) [reply onnxruntime/core/providers/cpu/signal/window_functions.cc:114]: Ok - Will do at all the places, and reply you.

Comment (madurais) [reply onnxruntime/core/providers/cpu/signal/window_functions.cc:115]: Hi @skottmckay  Pushed the changes with comment where hz1 is used.

**Review by skottmckay — APPROVED** [review]

**Pull request merged**